google/cloud/storage/internal/curl_client.cc

```diff
@@ -536,7 +536,9 @@ StatusOr<ObjectMetadata> CurlClient::InsertObjectMedia(
   // uploads. `DisableMD5Hash` and `DisableCrc32cChecksum` should not be
   // dependent on each other.
   if (!request.GetOption<DisableMD5Hash>().value() ||
-      !request.GetOption<DisableCrc32cChecksum>().value_or(false)) {
+      !request.GetOption<DisableCrc32cChecksum>().value_or(false) ||
```

Comment:
For `DisableCrc32cChecksum` we check `.value_or(false)`, but for `DisableMD5Hash` we just check `.value()`. Is that difference intentional?

Reply:
Yes, but a bit weird.  `DisableMD5Hash` is always set, its default constructor sets its value to `true` whereas `DisableCrc32cChecksum` defaults the value to `absl::nullopt`... groan.
I will clean this up in a future PR (#7029)